nats.go

```diff
@@ -1553,6 +1553,9 @@ func (nc *Conn) processConnectInit() error {
 		if nc.ptmr == nil {
 			nc.ptmr = time.AfterFunc(nc.Opts.PingInterval, nc.processPingTimer)
 		} else {
+			if !nc.ptmr.Stop() {
```

Comment:
With your approach, and a test where the client connects to a server, gets disconnected and connects again, we would be here and Stop() always returns false, and so we end-up in <-nc.ptmr.C that then blocks. Not sure if Go changed something on that regard, but I get a lot of Stop()->false, even when the callback has not be invoked yet.

If we set the timer to nil in stopPingTimer(), then this whole else can be removed because nc.ptmr will always be nil.